waiter/src/waiter/util/utils.clj

```diff
@@ -235,6 +236,21 @@
       (attach-waiter-namespace-keys data-map)
       (attach-waiter-source))))
 
+(let [b64-decoder (java.util.Base64/getUrlDecoder)]
+  (defn b64-url-json-decode [^String json-str]
+    "Decode a URL-safe Base64 JSON string into a Clojure data structure."
+    (-> (.decode b64-decoder json-str)
+        (String. java.nio.charset.StandardCharsets/UTF_8)
```

Comment:
Nitpick: import the `StandardCharsets` class.

Reply:
It was already imported for a different function! 😅
Dropped the package name.